### PR TITLE
Method for optimizing data structure functions

### PR DESCRIPTION
**Pull request by ut003460**
1.The extend() method was used to add the entire range of values at once, rather than adding them one by one. This can reduce the number of loops and the creation of temporary lists

### CODE REVIEW

File: avocado/utils/data_structures.py [hunk 1]
```diff
@@ -136,13 +136,13 @@ def comma_separated_ranges_to_list(string):
...
     values = []
-    for value in string.split(","):
-        if "-" in value:
-            start, end = value.split("-")
-            for val in range(int(start), int(end) + 1):
-                values.append(int(val))
+    ranges = string.split(",")
+    for range_str in ranges:
+        if "-" in range_str:
+            start, end = map(int, range_str.split("-"))
```
[the remark this answers is not in the document]
Comment (ut003460) [reply avocado/utils/data_structures.py]: Okay, thank you for your reply. The code has been revised again

**Comment by richtja** [comment]
Hi @ut003460 thank you for the update to the code LGTM. Can you please squash the `update` commit to the previous one? Thank you. 

You can do it with `git rebase -i HEAD~1`  and then mark the `update` commit as `fixup`

**Review by richtja — APPROVED** [review]
@ut003460 thank you.

**Pull request merged**